hyacinth_publish_target_clio_sync.rb

```diff
@@ -11,228 +11,308 @@
-	dynamic_field_data = digital_object_data['dynamic_field_data']
+    # If the full title begins with and ends with quotation marks, remove those quotation marks from the sort and nonsort portions
+    if title.match(/^".+"$/)
+      if title_non_sort_portion.length > 0
+        title_non_sort_portion = title_non_sort_portion[1...(title_non_sort_portion.length)]
+      else
+        title_sort_portion = title_sort_portion[1...(title_sort_portion.length)]
+      end
+      title_sort_portion = title_sort_portion[0...(title_sort_portion.length-1)]
+    end
+
+    puts "Num nonsort chars: #{num_nonsort_characters}" if debug
+    puts "Title Non-Sort Portion: #{title_non_sort_portion}" if debug
+    puts "Title Sort Portion: #{title_sort_portion}" if debug
+
+    # For some records, we want to pull 245 $b ("remainder of title") into the title too (because $a doesn't include all of our preferred keywords for search/display)
+    if include_245b && !marc_record['245']['b'].nil?
+      title_sort_portion += ' : ' + marc_record['245']['b'].gsub(/[\.\/,:]$/, '').strip # Remove certain trailing characters from the end of the "remainder of title"
+      if debug
+        puts '---'
+        puts '245 $b title found.  Appending to title_sort_portion...'
+        puts "New title_sort_portion value: #{title_sort_portion}"
+      end
+    end
+
+    # If part title is present, append it to end of title_sort_portion
+    unless marc_record['245']['p'].nil?
+      title_sort_portion += ': ' + marc_record['245']['p']
+      if debug
+        puts '---'
+        puts 'Part title found.  Appending to title_sort_portion...'
+        puts "New title_sort_portion value: #{title_sort_portion}"
+      end
+    end
+    vals = {
+      title: [
+          {
+            title_non_sort_portion: title_non_sort_portion,
+            title_sort_portion: title_sort_portion
+          }
+      ]
+    }
+    digital_object_data[:dynamic_field_data].merge! vals
+    unless marc_record['246'].nil?
+      if indicators?(marc_record['246'], '1', '3')
+        vals = {
+          alternative_title: [
+              {
+                alternative_title_value: marc_record['246']['a']
+              }
+          ]
+        }
+        digital_object_data[:dynamic_field_data].merge! vals
+      end
+    end
+    # URL/Location processing
+    marc_record.each_by_tag('856') do |field|
+      # 856 40 u resolver url/permanent link
+      if indicators?(field, '4', '0')
+        vals = { site_url: field['u'] }
+        digital_object_data[:publish_target_data].merge! vals
+      end
+      # 856 42 is related urls
+      if indicators?(field, '4', '2')
+        digital_object_data[:dynamic_field_data][:url] ||= []
+        digital_object_data[:dynamic_field_data][:url] << {
+          url_value: field['u'],
+          url_display_label: field['y'] || 'Related Electronic Resource'
+        }
+      end
+    # 920 is the site URL, and not processed for Hyacinth
```

Comment:
Do we need this comment?

Reply:
I wanted to hold onto it for documentary purposes for now, because it seemed important to OSMC to note it?